src/app/chip_data_model.gni

```diff
@@ -58,6 +58,47 @@ template("chip_data_model") {
     }
   }
 
+  chip_zapgen("${_data_model_name}_zapgen") {
+    input = rebase_path(invoker.zap_file)
+    generator = "app-templates"
+
+    outputs = [
+      "zap-generated/access.h",
+      "zap-generated/gen_config.h",
+      "zap-generated/endpoint_config.h",
+      "zap-generated/CHIPClientCallbacks.h",
+    ]
+
+    if (chip_code_pre_generated_directory == "") {
+      prune_outputs = []
+    }
+
+    if (chip_controller) {
+      outputs += [ "zap-generated/CHIPClusters.h" ]
+    } else {
+      if (defined(prune_outputs)) {
+        prune_outputs += [ "zap-generated/CHIPClusters.h" ]
+      }
+    }
+
+    # TODO: this logic seems wrong ...
```

Comment:
Please file an issue, since we have no more TODO bot.

Reply:
I think I can only add an issue once this code lands, since it would reference that this logic does not seem to have a clear explanation of why the ourputs and prune_outputs is acually defined.

Reply:
I added a better comment of why this seemed odd.

I generally preserved logic that existed before, but the statement of "if you have a pregenerated directory, this means that IMCommandHandler.cpp needs to be compiled in" is odd. I think a separate variable should be used, but unsurewhat it should be named - I dislike compile_command_handler because it does not seem to explain what builds compile the command handler in and I hope we can get something better in naming,

Reply:
Thank you, this is much better!